### PR TITLE
libcoap: some fixes for debugging enabled

### PR DESCRIPTION
**Pull request by OlegHahm**
Will also try to get them upstream.

### CODE REVIEW

File: pkg/libcoap/0006-fix-option-parsing-with-debug-enabled.patch [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
++++ b/option.c
+@@ -57,9 +57,10 @@ coap_opt_parse(const coap_opt_t *opt, size_t length, coap_option_t *result) {
+ 
+   switch(result->delta) {
+   case 15:
+-    if (*opt != COAP_PAYLOAD_START)
++    if (*opt != COAP_PAYLOAD_START) {
+       debug("ignored reserved option delta 15\n");
+-    return 0;
++      return 0;
```
Comment (gebart) [inline pkg/libcoap/0006-fix-option-parsing-with-debug-enabled.patch]: This seem like a fix for something, but I don't understand what without digging through the code, should this change maybe be upstreamed separately?

File: pkg/libcoap/0007-fix-debug-out-in-PDU-parsing.patch [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+@@ -930,10 +930,11 @@ if (!coap_pdu_parse((unsigned char *)buf, bytes_read, node->pdu)) {
+ #endif
+     unsigned char addr[INET6_ADDRSTRLEN+8];
+ 
+-    if (coap_print_addr(&src, addr, INET6_ADDRSTRLEN+8))
++    if (coap_print_addr(&src, addr, INET6_ADDRSTRLEN+8)) {
+       debug("** received %d bytes from %s:\n", (int)bytes_read, addr);
+ 
+-    coap_show_pdu( node->pdu );
++      coap_show_pdu( node->pdu );
```
Comment (gebart) [inline pkg/libcoap/0007-fix-debug-out-in-PDU-parsing.patch]: this also seem like a fix for something else.

**Comment by gebart** [comment]
the NDEBUG changes look good, could you split the non-NDEBUG changes to a separate commit and try to get it all upstreamed?

**Comment by gebart** [comment]
ACK for the whole thing after splitting the unrelated fixes into separate commits, or an explanation why they are related to NDEBUG.

**Comment by gebart** [comment]
As I found out in https://github.com/RIOT-OS/RIOT/pull/3319#issuecomment-121059083, you can strip the trailing whitespaces since `--ignore-whitespace` is given to `git am` in the Makefile

File: pkg/libcoap/0006-fix-option-parsing-with-debug-enabled.patch (hunk 1 above)
Comment (OlegHahm) [inline pkg/libcoap/0006-fix-option-parsing-with-debug-enabled.patch]: Well, actually the question is if the curly brackets are missing on purpose or not. I'll open an issue at libcoap repository.

**Comment by gebart** [comment]
ping @OlegHahm

**Comment by OlegHahm** [comment]
Sorry, I wanted to get the fixes upstream first. Let me check, what I did.

**Comment by OlegHahm** [comment]
Ok synchronized the two additional fixes with libcoap master and split into separate commits. Does your ACK hold, @gebart?

**Comment by OlegHahm** [comment]
Apparently there is still an issue with whitespaces.

**Comment by OlegHahm** [comment]
Should be fixed now.

**Comment by gebart** [comment]
ACK holds

**Comment by gebart** [comment]
though you need to fix the whitespace errors.
"space before tab in indent"
run a `git diff --check master` on you branch

**Comment by OlegHahm** [comment]
I had removed the tab in a previous version, but `git am` was complaining - even with --ignore-whitespaces and the like.

**Comment by OlegHahm** [comment]
Ah, now I see what is the problem. But I wonder what to do about it. It's basically code that we take unchanged from libcoap.

**Comment by OlegHahm** [comment]
This seems to fix it.

**Comment by OlegHahm** [comment]
fixed enumeration

**Comment by OlegHahm** [comment]
@gebart, okay, to squash?

**Comment by gebart** [comment]
OK to squash, ACK.

Travis seem fine now except for the squashing.

**Comment by OlegHahm** [comment]
Travis is green.

**Pull request merged**